### ISSUE TITLE
CreateSwapChain() Issue after the latest update Novermber 15

### ISSUE THREAD

**Issue by lostdevfound** [ValveSoftware/Dota-2#2603]
Dota 2 stopped working after the latest update.

When launching Dota 2 I get:
```CreateSwapChain():
The selected graphics queue does not support presenting a swapping image
```

Can you guys update the game and not break it?
Every update, there is some new bug on linux.

Since, the *** Compenium, I experience crashes when party queuing and now dota doesn't run at all.

#### Your system information
Steam Client Build Date:  Mon, Oct 30 10:09 PM UTC -08:00
Steam Web Build Date:  Fri, Oct 27 7:29 PM UTC -08:00
Steam API Version:  SteamClient020

Computer Information:
Manufacturer: Micro-Star International Co., Ltd.
Model: PRO Z690-A WIFI DDR4(MS-7D25)
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: GenuineIntel
CPU Brand: 12th Gen Intel(R) Core(TM) i7-12700K
CPU Family: 0x6
CPU Model: 0x97
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 4900 MHz
20 logical processors
12 physical processors
Hyper-threading: Supported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Unsupported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Unsupported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Unsupported
AVX512VNNI: Unsupported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
Ubuntu 22.04.3 LTS (64 bit)
Kernel Name: Linux
Kernel Version: 6.2.0-36-generic
X Server Vendor: The X.Org Foundation
X Server Release: 12101004
X Window Manager: GNOME Shell
Steam Runtime Version: steam-runtime_0.20231024.64411
Video Card:
Driver: NVIDIA Corporation NVIDIA GeForce GTX 1660 SUPER/PCIe/SSE2
Driver Version: 4.6.0 NVIDIA 525.147.05
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID: 0x10de
DeviceID: 0x21c4
Revision Not Detected
Number of Monitors: 1
Number of Logical Video Cards: 1
Primary Display Resolution: 1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 20.79" x 11.69" (23.82" diag), 52.8cm x 29.7cm (60.5cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 6144 MB
Supported MSAA Modes: 2x 4x 8x 16x
Sound card:
Audio device: Realtek ALC897
Memory:
RAM: 31901 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: en_CA.UTF-8
Total Hard Disk Space Available: 937291 MB
Largest Free Hard Disk Block: 635908 MB
Storage:
Number of SSDs: 2
SSD sizes: 4000G,1000G
Number of HDDs: 0
Number of removable drives: 0

**Comment by vincentrinot** [comment]
I am having the same issue, first time ever seeing this error before. Just after the update.

**Comment by UtkarshDixit-97** [comment]
Same here in my debian 12 , kde plasma , wayland

If i start from terminal by manually going to dota installation folder and running ./dota.sh, I get

Loaded /home/user/Documents/user - Usage/Steam Library/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x558d25d6ee50
fcntl(7) for /tmp/source_engine_2808995433.lock failed
Loaded libSDL3.so.0, got 0x558d25db24f0
zenity: symbol lookup error: /lib/x86_64-linux-gnu/libharfbuzz.so.0: undefined symbol: FT_Get_Transform

**Comment by agrahamlincoln** [comment]
![dota-error](https://github.com/ValveSoftware/Dota-2/assets/8289990/017f043f-27c4-43b6-a10f-096036ea5056)

same thing here, I was in game immediately before the patch

**Comment by lostdevfound** [comment]
I tried updating my nvidia drivers from 525 to 535. This did not help.
I also tried various launch settings like `-vulkan_disable_graphics_pipeline_library` or `-vulakn -vulkan_disable_graphics_pipeline_library`. This also didn't help.

**Comment by timmac41** [comment]
Same issue:
Arch 6.6.1-arch1-1
KDE
Wayland
RX6800XT
i5-9600K

![image](https://github.com/ValveSoftware/Dota-2/assets/147016710/18eb6a2c-4758-494d-8d88-0f0a525e19de)

**Comment by roboconcept** [comment]
Same issue as @timmac41 w/ Ubuntu 22.04 on AMD Radeon 5600xt

**Comment by danielgural** [comment]
Same on Ubuntu 22, update nvidia drivers, vulkan drivers no luck

**Comment by arielnmz** [comment]
Can confirm too
Fedora release 37 (Thirty Seven)
AwesomeWM
X11
RX6600XT
R7-5800X

Both native steam (RPM) and flatpak, although the exact issue is different:

Flatpak:
```
CreateSwapChain():
Failed to create VkSurfaceKHR.
```

Native:
```
CreateSwapChain():
The selected graphics queue does not support presenting a swapchain image.
```

**Comment by Chaliuz** [comment]
I got the same problem, I installed experimental proton in library > dota > properties > compatibility  and dota runs fine.

**Comment by josechudev** [comment]
Same issue here:

"EndeavourOS Linux" (64 bit)
Kernel Name: Linux
Kernel Version: 6.6.1-arch1-1
X Server Vendor: The X.Org Foundation
X Server Release: 12101009
X Window Manager: KWin
Steam Runtime Version: steam-runtime_0.20231024.64411

**Comment by agrahamlincoln** [comment]
> I got the same problem, I installed experimental proton in library > dota > properties > compatibility and dota runs fine.

I tried proton experimental, the game opens to the splash screen and immediately crashes

**Comment by dire-alpaca** [comment]
I get the same error. Seems to seg fault with this last message.
`Failed to create VkSurfaceKHR. `

Arch Linux
KDE Plasma Wayland and X11 
Linux 6.6.1-arch1-1
Mesa 23.2.1

AMD Ryzen 5950x 
32Gb Ram
AMD 6800xt

**Comment by vincentrinot** [comment]
> > I got the same problem, I installed experimental proton in library > dota > properties > compatibility and dota runs fine.
> 
> I tried proton experimental, the game opens to the splash screen and immediately crashes

i thought i was gonna share in here a successful launch with experimental proton. Same thing happened to me lmao

**Comment by AungKoKoLin1997** [comment]
I got same experience in ubuntu 22.04.

**Comment by zeroedout** [comment]
It launches with Proton for me and bot matches work but you cannot connect to live games, neither playing nor watching.

**Comment by pleish** [comment]
Same. Cannot launch after latest patch.

**Comment by gachiemchiep** [comment]
same problem.
try switching into Proton but can't play

**Comment by Chaliuz** [comment]
> It launches with Proton for me and bot matches work but you cannot connect to live games, neither playing nor watching.

same :(

**Comment by Niflheimr-dog** [comment]
Hello, I have the same problem, I can’t start the game after the update, it says an error
`The selected graphics queue does not support presenting a swapping image`
although yesterday night I launched through a volcano and everything worked properly, whoever found a solution, please help.

I'm adding an error code for launching through the terminal, maybe it will help

`./dota.sh: line 122: 11688 Segmentation fault      (core dumped) ${STEAM_RUNTIME_PREFIX} ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"`

P.s: I tried launching through "Proton Experimental" but only the screen saver starts and then crashes, and so on with all versions of Proton

**Comment by YangtseSu** [comment]
Another discussion:
https://steamcommunity.com/app/570/discussions/0/6027566653451436967/?ctp=2

**Comment by YangtseSu** [comment]
The developer didn't even run it on Linux once before releasing the update.

**Comment by UtkarshDixit-97** [comment]
> The developer didn't even run it on Linux once before releasing the update.

Seems like exactly this is what has happened. And yet you are receiving down thumbs. Hmm

isn't working on any of my 3 linux machines, all with different distros.

**Comment by stryan** [comment]
> Seems like exactly this is what has happened. And yet you are receiving down thumbs. Hmm

This is an issue tracker.  If you're going to complain at least post a system specs gist and other details to actually help with the discussion. Save the "valve doesn't care" etc posts for reddit or something.

**Comment by andrelbc** [comment]
Can't launch the game after the last update, but here I don't get any error popup, when I try to launch through the terminal then I get the Segmentation fault error. With proton, I can get to the menu, but then it crashes right away.

**Comment by arielnmz** [comment]
@stryan this is not a private bug tracker and is open to the community for a reason, and their comment perfectly summarizes the community's feeling about the matter at the very least. Go boohoo somewhere else dude

**Comment by AungKoKoLin1997** [comment]
> this is not a private bug tracker and is open to the community for a reason, and their comment perfectly summarizes the community's feeling about the matter at the very least. Go boohoo somewhere else dude

I agree with you. Developer didn't actually test in every OS before it is launched. This is basic thing as a developer. I really surprised and QA team didn't recognize this bug.

**Comment by Epic-R-R** [comment]
Same here in my Ubuntu 22.04, after update game

**Comment by arielnmz** [comment]
Ppl in this thread really be like 
![image](https://github.com/ValveSoftware/Dota-2/assets/24735547/befe099b-d428-42f1-a73f-7838c8b427a1)

**Comment by TheBicPen** [comment]
Same issue here. Wayland, KDE Neon, GTX 1060. [Specs](https://gist.github.com/TheBicPen/6b30bd23d1a36b733cde467dd3d64ec9)

**Comment by YangtseSu** [comment]
> > Seems like exactly this is what has happened. And yet you are receiving down thumbs. Hmm
> 
> This is an issue tracker. If you're going to complain at least post a system specs gist and other details to actually help with the discussion. Save the "valve doesn't care" etc posts for reddit or something.

I couldn't agree more with you. I'm just making suggestions about the testing process before an update is released.
The system specs are unnecessary on this issue. The only useful information is the OS. As long as it is Linux, this bug can be reproduced.

**Comment by Strykar** [comment]
Confirming same issue on a GTX 1080 on Arch Linux.

**Comment by disnaut** [comment]
Confirming same issue on a 7900xtx on Pop-OS

**Comment by Blindinlights** [comment]
Confirming same issue on a RTX 3050 on Arch Linux.

**Comment by alou-S** [comment]
This seems like an average Valve fucks up the libraries moment. Will be fixed soon hopefully. Will try for a workaround later.

Things like this has happened multiple times before on CS2 and dota2

**Comment by robertkirkman** [comment]
> This seems like an average Valve fucks up the libraries moment. Will be fixed soon hopefully. Will try for a workaround later.

i think it's a problem inside of `dota 2 beta/game/bin/linuxsteamrt64/dota2` because it was modified in the update and it generates the error window so I think only Valve is capable of fixing it by editing the code of that file and recompiling it

**Comment by ulastanersoyak** [comment]
same in gtx1650

**Comment by hopprJoseMariLumanlan** [comment]
Please fix it valved. You're losing the linux gamers! Hahaha I just want to play and have a good time.

Hope this helps!
![image](https://github.com/ValveSoftware/Dota-2/assets/64821085/86139fc4-5c93-4792-a768-0e23a053a303)

**Comment by earn9** [comment]
same dota issue on popos... unsubscribed. no testing before patch release?!

**Comment by rodentskie** [comment]
Same issue.

![image](https://github.com/ValveSoftware/Dota-2/assets/49242131/69757499-469e-4dba-974e-5eb284841723)

**Comment by alou-S** [comment]
A temporary way to get the game running for now would be to run it through Proton Experimental. 

**This has a good chance to cause other issues so only use it only if you are desperate**

**Comment by waylon531** [comment]
Somehow the update broke proton as well ... it won't even launch for me any more on proton

**Comment by robertkirkman** [comment]
> A temporary way to get the game running for now would be to run it through Proton Experimental.
> 
> **This has a good chance to cause other issues so only use it only if you are desperate**

When I use any Proton version I see the logo for 1 second and then the game crashes. If I use `%command% -novid` in combination with that, I see the main menu for 1 second and then the game crashes

**Comment by bzk621** [comment]
same issue
![image](https://github.com/ValveSoftware/Dota-2/assets/25078988/43b86bb3-8d6b-4cdb-b168-cb583c1ff642)

**Comment by bzk621** [comment]
I thought I was only one with issue, but it turns out I’m not only one.

**Comment by bzk621** [comment]
With what parameters do you launch game with [Launch Options](https://dota2.fandom.com/wiki/Launch_Options)?

**Comment by Dorjpalam** [comment]
![Screenshot from 2023-11-16 15-31-11](https://github.com/ValveSoftware/Dota-2/assets/6973531/92679404-94e5-4337-b31b-a6ffecfcc634)

Exact same issue occured.

**Comment by bzk621** [comment]
Message from Steam Support on Nov 16 @ 2:37pm | 11 minutes ago
Hello there,

I'm sorry to hear what happened.

The team is aware that some users are unable to run Dota on Steam Deck since "The International 2023 Celebration" update.

Unfortunately, we do not have a solution to suggest at this time, as this will likely need a development-side fix. Please keep an eye on the Dota Updates blog as the game is updated frequently, and when this issue is fixed it will likely be included in the patch notes there

**Comment by VladislavNekto** [comment]
@bzk621, It's obvious that Steam support won't help; if there were a solution, we would have received it by now. Perhaps all that's left is to wait for a hotfix (hot as valvetime).

**Comment by zekebaker** [comment]
Mint here, same issue.
Love you Linux Gamers.

**Comment by MrSirKingLord** [comment]
Same issue

![image](https://github.com/ValveSoftware/Dota-2/assets/92161916/8d705195-3cf6-4af5-9293-5fab0d8455be)


OS: LMDE 6 (faye) x86_64
Kernel: 6.1.0-13-amd64
GPU: NVIDIA GeForce RTX 3080
nvidia-driver/stable,now 525.125.06-1~deb12u1 amd64 [installed]

**Comment by dohaicuong** [comment]
same issue

![image](https://github.com/ValveSoftware/Dota-2/assets/20808725/13614a46-bbd5-4a23-8efa-62272b70c88e)

**Comment by florinDNL** [comment]
+1 

![image](https://github.com/ValveSoftware/Dota-2/assets/79944491/3d8fddd0-3a74-43f2-983e-b270cbbd558c)

**Comment by ong-andrew** [comment]
Same issue. 

When trying to launch, this pops up:

CreateSwapChange():
The selected graphic queue does not support presenting a swapchain image

ps. Please stop posting Neofetch. This is a issues tracker. Devs don't need to know if youave 64gb ram and i9 chip. Just post what is relevant.

ps. much love to linux port devs and gamers.

**Comment by dmknght** [comment]
> > this is not a private bug tracker and is open to the community for a reason, and their comment perfectly summarizes the community's feeling about the matter at the very least. Go boohoo somewhere else dude
> 
> I agree with you. Developer didn't actually test in every OS before it is launched. This is basic thing as a developer. I really surprised and QA team didn't recognize this bug.

and they managed to break Dota 2 startup before the TI. How is that possible? I meant random crashes while playing game, yeah that makes sense. But this is game just can't start.

**Comment by staticf0x** [comment]
I noticed two things:

1. `zenity: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory` - However, copying one from my `/usr/lib64` doesn't work as it uses `/usr/lib64/libharfbuzz.so` but some symbol is not there: `FT_Get_Color_Glyph_Layer`.
2. It doesn't help to use `-gl` or `-safe_mode`. I thought it would fall back to OpenGL and thus avoid Vulkan, but it doesn't seem so.

**Comment by dmknght** [comment]
> I noticed two things:
> 
>     1. `zenity: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory` - However, copying one from my `/usr/lib64` doesn't work as it uses `/usr/lib64/libharfbuzz.so` but some symbol is not there: `FT_Get_Color_Glyph_Layer`.
> 
>     2. It doesn't help to use `-gl` or `-safe_mode`. I thought it would fall back to OpenGL and thus avoid Vulkan, but it doesn't seem so.

The `1.` looks like the version that Dota 2 is using (latest patch) is different, therefore symbol (or function?) `FT_Get_Color_Glyph_Layer` doesn't exist. Regarding the `2.`, I think Dota 2 completely dropped OpenGL?

**Comment by staticf0x** [comment]
> Regarding the `2.`, I think Dota 2 completely dropped OpenGL?

Ah I think you're right, I missed that. Here's a comment from Valve that OpenGL is no longer supported: https://github.com/ValveSoftware/Dota-2/issues/1983#issuecomment-1566119402

Edit: and [dota2.com announcement](https://www.dota2.com/newsentry/3659774959159800240)

**Comment by Niflheimr-dog** [comment]
> Я заметил две вещи:
> 
> 1. `zenity: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory`- Однако копирование одного из моих `/usr/lib64`не работает, поскольку он использует `/usr/lib64/libharfbuzz.so`, но какого-то символа там нет: `FT_Get_Color_Glyph_Layer`.
> 2. Использование `-gl`или `-safe_mode`. Я думал, что придется вернуться к OpenGL и, таким образом, избежать Vulkan, но это не так.

My hope Valvе will see this...

**Comment by thevivotran** [comment]
I am facing the same issue. I have already:
- Purged uninstall nvidia driver, then reinstalled the recommended one
- Removed and reinstalled the whole game

The following image is my PC specs.
![image](https://github.com/ValveSoftware/Dota-2/assets/68809439/94ca720c-c910-41ee-bdff-489ff3d3e6aa)

**Comment by DreadKnight** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814144105

Good to know you tried that stuff, saves me some trouble 🐻

**Comment by alleey** [comment]
Slightly different desc here (Ubuntu 23). **And totally agree with the game crashing and other weird stuff after the compendium update**.

![image](https://github.com/ValveSoftware/Dota-2/assets/2000085/df80bdce-928f-4b54-af09-7ecf8e58d813)

**Comment by the-illarionov** [comment]
I guess we just have to wait for morning to come to Central America

**Comment by slovak12345** [comment]
I also have such error in PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

**Comment by endoft1me** [comment]
Same issue here.
![Screenshot from 2023-11-16 11-54-22](https://github.com/ValveSoftware/Dota-2/assets/151025564/99c8955c-090a-44ad-9303-f514ac89905f)

**Comment by faroperfect** [comment]
...
![Uploading ScreenShots_dota_boot_errorText.png…]()

**Comment by ahmedgadit** [comment]
Not working Ubuntu 20.04 as well same configuration as above mentioned in issue sharing details related to ubuntu 20.04, it was running last night before the update 

### Operating System Version:
Ubuntu 20.04.6 LTS (64 bit)
Kernel Name: Linux
Kernel Version: 5.15.0-88-generic
X Server Vendor: The X.Org Foundation
X Server Release: 12013000
X Window Manager: GNOME Shell
Steam Runtime Version: steam-runtime_0.20231024.64411
### Video Card:
Driver: AMD AMD KAVERI (DRM 3.42.0, 5.15.0-88-generic, LLVM 12.0.0)
Driver Version: 4.6 (Compatibility Profile) Mesa 21.2.6
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 59 Hz
VendorID: 0x1002
DeviceID: 0x130f
Revision Not Detected
Number of Monitors: 1
Number of Logical Video Cards: 1
Primary Display Resolution: 1600 x 900
Desktop Resolution: 1600 x 900
Primary Display Size: 17.44" x 9.80" (20.00" diag), 44.3cm x 24.9cm (50.8cm diag)
Primary VRAM: 1024 MB

**Comment by rishabharoraa** [comment]
Facing the same issue here
![image](https://github.com/ValveSoftware/Dota-2/assets/43946451/b7b0772c-69d6-4b6f-ab6a-3398538d7b17)
![image](https://github.com/ValveSoftware/Dota-2/assets/43946451/ccf09cef-6836-4271-9b5f-5efcf9d8776e)

**Comment by AriePriyambadha** [comment]
![Screenshot from 2023-11-16 18-13-31](https://github.com/ValveSoftware/Dota-2/assets/6871368/224074d7-3c8e-458d-9909-2bde4759ea9e)

![Screenshot from 2023-11-16 18-14-21](https://github.com/ValveSoftware/Dota-2/assets/6871368/58614182-3238-4d3e-89fc-260011e3ce5e)

I have the same problem too

**Comment by gentuser** [comment]
Same problem on kernel 6.6.1, ubuntu 23.10, Nvidia 4080, and driver 525.147.05.
Attached is the crash dump.
[crash_20231116121445_2.zip](https://github.com/ValveSoftware/Dota-2/files/13376939/crash_20231116121445_2.zip)

**Comment by Hursofid** [comment]
Same here

```
       _,met$$$$$gg.          constructed@workstation 
    ,g$$$$$$$$$$$$$$$P.       ----------------------- 
  ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 11 (bullseye) x86_64 
 ,$$P'              `$$$.     Host: A320M-H 
',$$P       ,ggs.     `$$b:   Kernel: 5.10.0-26-amd64 
`d$$'     ,$P"'   .    $$$    Uptime: 1 day, 1 hour, 51 mins 
 $$P      d$'     ,    $$P    Packages: 2321 (dpkg), 133 (brew) 
 $$:      $$.   -    ,d$$'    Shell: bash 5.2.21 
 $$;      Y$b._   _,d$P'      Resolution: 1440x2560, 2560x1440 
 Y$$.    `.`"Y$$$$P"'         DE: GNOME 3.38.6 
 `$$b      "-.__              WM: Mutter 
  `Y$$                        WM Theme: Adwaita 
   `Y$$.                      Theme: Adwaita-dark [GTK2/3] 
     `$$b.                    Icons: Adwaita [GTK2/3] 
       `Y$$b.                 Terminal: gnome-terminal 
          `"Y$b._             CPU: AMD Ryzen 5 3600 (12) @ 3.600GHz 
              `"""            GPU: AMD ATI Radeon RX 470/480/570/570X/580/580X/590 
                              Memory: 9703MiB / 32081MiB 
```
Some logs:
```
Nov 16 12:32:18 workstation steam.desktop[51452]: /bin/sh\0-c\0SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0 /home/constructed/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/constructed/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/games/steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/games/steam/steamapps/common/dota 2 beta/game/dota.sh' +@panorama_min_comp_layer_dimension 0 -prewarm_panorama -novid -map dota -high -fullscreen -phased_window_create -x 1921 -w 2560 -h 1440 -sdl_displayindex 1 +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0
Nov 16 12:32:18 workstation steam.desktop[52228]: ERROR: ld.so: object '/home/constructed/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
[ Duplicates truncated ]
Nov 16 12:32:18 workstation steam.desktop[52231]: ERROR: ld.so: object '/home/constructed/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Nov 16 12:32:19 workstation steam.desktop[52345]: pid 52345 != 52344, skipping destruction (fork without exec?)
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x563a8c184c20
Nov 16 12:32:19 workstation steam.desktop[52348]: Using breakpad crash handler
Nov 16 12:32:19 workstation steam.desktop[52348]: [S_API] SteamAPI_Init(): Loaded '/home/constructed/.local/share/Steam/linux64/steamclient.so' OK.
Nov 16 12:32:19 workstation steam.desktop[52348]: Setting breakpad minidump AppID = 570
Nov 16 12:32:19 workstation steam.desktop[52348]: Forcing breakpad minidump interfaces to load
Nov 16 12:32:19 workstation steam.desktop[52348]: Looking up breakpad interfaces from steamclient
Nov 16 12:32:19 workstation steam.desktop[52348]: Calling BreakpadMiniDumpSystemInit
Nov 16 12:32:19 workstation steam.desktop[52348]: 11/16 12:32:19 Init: Installing breakpad exception handler for appid(570)/version(8500774)/tid(52348)
Nov 16 12:32:19 workstation steam.desktop[52348]: Looking up breakpad interfaces from steamclient
Nov 16 12:32:19 workstation steam.desktop[52348]: Calling BreakpadMiniDumpSystemInit
Nov 16 12:32:19 workstation steam.desktop[52348]: SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198028997990 [API loaded yes]
Nov 16 12:32:19 workstation steam.desktop[52348]: SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561198028997990
Nov 16 12:32:19 workstation steam.desktop[52348]: Setting breakpad minidump AppID = 373300
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded libSDL3.so.0, got 0x563a8c4641d0
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x563a8c16ebc0
Nov 16 12:32:19 workstation xdg-desktop-por[3628]: Backend call failed: GDBus.Error:org.gnome.SessionManager.GeneralError: Application ID not specified
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x563a8c4aec00
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x563a8c4d8c00
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x563a8c554fe0
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x563a8c544080
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x563a8c5758c0
Nov 16 12:32:19 workstation steam.desktop[52348]: Loaded /games/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x563a8c575100
Nov 16 12:32:23 workstation gnome-shell[1785]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x5e00211
Nov 16 12:32:25 workstation steam.desktop[52412]: crash_20231116123225_2.dmp[52412]: Uploading dump (out-of-process)
Nov 16 12:32:25 workstation steam.desktop[52412]: /tmp/dumps/crash_20231116123225_2.dmp
Nov 16 12:32:25 workstation kernel: dota2[52348]: segfault at 0 ip 00007fa7bfc2f526 sp 00007ffe045a94f0 error 6 in libtier0.so[7fa7bfb77000+2f3000]
Nov 16 12:32:25 workstation kernel: Code: 2d c6 f3 24 00 31 db 0f 1f 44 00 00 49 8b 74 dc 08 48 85 f6 74 0a 4c 89 ef 31 c0 e8 84 d8 f4 ff 48 83 c3 01 41 39 1c 24 7f e2 <c7> 04 25 00 00 00 00 00 00 00 00 0f 0b c7 04 25 00 00 00 00 00 00
Nov 16 12:32:25 workstation steam.desktop[52344]: /games/steam/steamapps/common/dota 2 beta/game/dota.sh: line 122: 52348 Segmentation fault      ${STEAM_RUNTIME_PREFIX} ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
Nov 16 12:32:27 workstation steam.desktop[52412]: crash_20231116123225_2.dmp[52412]: Finished uploading minidump (out-of-process): success = yes
Nov 16 12:32:27 workstation steam.desktop[52412]: crash_20231116123225_2.dmp[52412]: response: CrashID=bp-79b05be1-593a-4d12-8bda-da9a22231116
Nov 16 12:32:27 workstation steam.desktop[52412]: crash_20231116123225_2.dmp[52412]: file ''/tmp/dumps/crash_20231116123225_2.dmp'', upload yes: ''CrashID=bp-79b05be1-593a-4d12-8bda-da9a22231116''
```

**Comment by neimovirne** [comment]
https://github.com/ValveSoftware/Dota2-Gameplay/issues/13555

**Comment by neimovirne** [comment]
same
#https://github.com/ValveSoftware/Dota2-Gameplay/issues/13555

**Comment by icf20** [comment]
https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527  2017 error

**Comment by Hursofid** [comment]
> [#1215 (comment)](https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527) 2017 error

Have you tried it yet?

**Comment by Zap1999** [comment]
> > [#1215 (comment)](https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527) 2017 error
> 
> Have you tried it yet?

I have found the file on Ubuntu 22.04 at `/usr/share/doc/xserver-xorg-video-intel/xorg.conf`. It didn't help though.

**Comment by ahmedgadit** [comment]
> [#1215 (comment)](https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527) 2017 error

Yes i have tried no xorg configuration file in latest ubuntu version, i am using mesa 21v so its old fix not to the current one, it was for glopen now valve doesnt support opengl  

@Zap1999 @Hursofid

**Comment by huhongjun** [comment]
Same issue here. ubuntu 22.04 + GTX 1060i

**Comment by neimovirne** [comment]
> [#1215 (comment)](https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527) 2017 error

by your logic every time valve breaks the game players have to dance with a tambourine to fiddle with the system files to get the game work again, even if it was fully functional a minute before the update that broke it?

**Comment by ReDTerraN** [comment]
Just reporting in to say I same issue here. Manjaro RTX 5500. I believe this is for most Linux users.

Please Valve, atleast have a CI pipeline test that checks on Linux if the game can run before pushing to prod.

**Comment by florinDNL** [comment]
> > [#1215 (comment)](https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527) 2017 error
> 
> by your logic every time valve breaks the game players have to dance with a tambourine to fiddle with the system files to get the game work again, even if it was fully functional a minute before the update that broke it?

I think at this point people are just trying to find ways to run the game since who knows how long it takes Valve to react

**Comment by AungKoKoLin1997** [comment]
It is almost 12 hours and value still doesn't fix it. Nice.

**Comment by alou-S** [comment]
> [#1215 (comment)](https://github.com/ValveSoftware/Dota-2/issues/1215#issuecomment-351301527) 2017 error

It isn't a 2017 error. There are many faults that can trigger a single error message. And its extremely unlikely that the fixes used in that is going to work.

**Comment by nikolaPantelic98** [comment]
Same for linux mint 21.1

**Comment by luria** [comment]
Same problem in Linux Mint!
 ![image](https://github.com/ValveSoftware/Dota-2/assets/13627989/cf43baf7-45a4-47f0-89cd-1b5eda85179f)

**Comment by TheCoolerSuptelov** [comment]
same problem on mint
![image](https://github.com/ValveSoftware/Dota-2/assets/102233851/f93569ba-ef45-4df1-b6c6-386d91edfead)

**Comment by josephlett** [comment]
i have the same problem with ubuntu nvidia 2060 rtx
what is the solution or do we just wait on a new dota update

**Comment by josephlett** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issue-1995902925

what is the solution?

**Comment by alou-S** [comment]
@josephlett When quoting a reply you don't need to quote the entire reply. Just the first 2 to three lines. It would be nice if you could edit your reply and remove the extra lines. You could probably instead add a link to the reply along with the first few lines for reference

**Comment by pure-bliss** [comment]
If we could stop spamming the issue with already known information, useless images of screenfetches and "same for me" comments maybe it would be easier to notice if some official and usefull response arrived. Please stop the "same" and other comments which don't add any value to the issue. It also makes the devs work way harder

**Comment by kuramayouko** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814337934

wait for devs to fix it

**Comment by Fristivan** [comment]
same problem on KDE NEON

**Comment by IronBlood** [comment]
My console said something like this:

```
/bin/sh\0-c\0/home/runner/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/runner/.local/share/Steam/ubuntu12_32/steam-launch-wrappe
r -- '/home/runner/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/runner/.local/share
/Steam/steamapps/common/dota 2 beta/game/dota.sh' +@panorama_min_comp_layer_dimension 0 -prewarm_panorama -perfectworld -novid -sdlaudiodriver pipewire\0     
chdir "/home/runner/.local/share/Steam/steamapps/common/dota 2 beta"                                                                                       
ERROR: ld.so: object '/home/runner/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32)
: ignored.                                                                                                                                                    
ERROR: ld.so: object '/home/runner/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64)
: ignored.                                                                                                                                                    
ERROR: ld.so: object '/home/runner/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32)
: ignored.                                                                                                                                                    
ERROR: ld.so: object '/home/runner/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32)
: ignored.                                                                                                                                                    
ERROR: ld.so: object '/home/runner/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32)
: ignored.                                                                                                                                                    
pressure-vessel-wrap[2061397]: W: Failed to load Vulkan ICD #0 from /etc/vulkan/icd.d/amd_icd32.json: Failed to open file '/etc/vulkan/icd.d/amd_icd32.json': 
No such file or directory                                                                                                                                     
pressure-vessel-wrap[2061397]: W: Failed to load Vulkan ICD #1 from /etc/vulkan/icd.d/amd_icd64.json: Failed to open file '/etc/vulkan/icd.d/amd_icd64.json': 
No such file or directory                                                                                                                                     
pressure-vessel-wrap[2061397]: W: Failed to load Vulkan implicit layer #0 from /etc/vulkan/implicit_layer.d/amd_icd32.json: Failed to open file '/etc/vulkan/i
mplicit_layer.d/amd_icd32.json': No such file or directory                                                                                                    
pressure-vessel-wrap[2061397]: W: Failed to load Vulkan implicit layer #1 from /etc/vulkan/implicit_layer.d/amd_icd64.json: Failed to open file '/etc/vulkan/i
mplicit_layer.d/amd_icd64.json': No such file or directory                                                                                                    
pid 2061537 != 2061536, skipping destruction (fork without exec?)
pid 2061538 != 2061536, skipping destruction (fork without exec?)
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x55ebdfa9a2f0
Using breakpad crash handler
[S_API] SteamAPI_Init(): Loaded '/home/runner/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
11/16 20:16:37 Init: Installing breakpad exception handler for appid(570)/version(8500774)/tid(2061542)
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198098917410 [API loaded yes]
SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561198098917410
Setting breakpad minidump AppID = 373300
Loaded libSDL3.so.0, got 0x55ebdfd4f9c0 
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x55ebdfa839b0
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x55ebdfd9a5b0
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x55ebdfd9aef0
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x55ebdfe0d550
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x55ebdfe08a70
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x55ebdfe25b30
Loaded /home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x55ebdfe26850
Fossilize INFO: Overriding serialization path: "/home/runner/.local/share/Steam/steamapps/shadercache/570/fozpipelinesv6/steamapprun_pipeline_cache".
crash_20231116201657_2.dmp[2061605]: Uploading dump (out-of-process)
/home/runner/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh: line 122: 2061542 Segmentation fault      (core dumped) ${STEAM_RUNTIME_PREFIX} 
${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
Game 570 created interface STEAMAPPLIST_INTERFACE_VERSION001 / 
Game 570 created interface STEAMAPPS_INTERFACE_VERSION008 / 
Game 570 created interface STEAMHTMLSURFACE_INTERFACE_VERSION_005 / 
Game 570 created interface STEAMHTTP_INTERFACE_VERSION003 / 
Game 570 created interface STEAMINVENTORY_INTERFACE_V003 / 
Game 570 created interface STEAMMUSICREMOTE_INTERFACE_VERSION001 / 
Game 570 created interface STEAMMUSIC_INTERFACE_VERSION001 / 
Game 570 created interface STEAMPARENTALSETTINGS_INTERFACE_VERSION001 / 
Game 570 created interface STEAMREMOTESTORAGE_INTERFACE_VERSION016 / 
Game 570 created interface STEAMSCREENSHOTS_INTERFACE_VERSION003 / 
Game 570 created interface STEAMUGC_INTERFACE_VERSION018 / 
Game 570 created interface STEAMUSERSTATS_INTERFACE_VERSION012 / 
Game 570 created interface STEAMVIDEO_INTERFACE_V003 / 
Game 570 created interface SteamController008 / 
Game 570 created interface SteamFriends017 / 
Game 570 created interface SteamInput006 / 
Game 570 created interface SteamMatchGameSearch001 / 
Game 570 created interface SteamMatchMaking009 / 
Game 570 created interface SteamMatchMakingServers002 / 
Game 570 created interface SteamNetworking006 / 
Game 570 created interface SteamUser023 / 
Game 570 created interface SteamUser023 / User
Game 570 created interface SteamUtils010 / 
Game 570 method call count for IClientAppManager::GetCurrentLanguage : 1
Game 570 method call count for IClientAppManager::GetAppStateInfo : 1
Game 570 method call count for IClientUtils::RecordSteamInterfaceCreation : 24
Game 570 method call count for IClientUtils::GetSteamUILanguage : 1
Game 570 method call count for IClientUtils::GetAppID : 27
Game 570 method call count for IClientFriends::GetFriendPersonaName_Public : 2
Game 570 method call count for IClientUser::GetUserDataFolder : 1
Game 570 method call count for IClientUser::GetSteamID : 5
Uploaded AppInterfaceStats to Steam
```

Ubuntu 22.04 / 6.2.0-36-generic / NVIDIA 535.129.03

**Comment by Newbytee** [comment]
Not a fix per se, but adding `env -u SDL_VIDEODRIVER %command%` to the launch options gives me a black fullscreen window with the title "Dota 2" before the error appears instead of just the error message.

**Comment by MrFox131** [comment]
```
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x560297c4de10
Using breakpad crash handler
[S_API] SteamAPI_Init(): Loaded '/home//******/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
11/16 22:38:44 Init: Installing breakpad exception handler for appid(570)/version(8500774)/tid(254633)
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198333115652 [API loaded yes]
SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561198333115652
Setting breakpad minidump AppID = 373300
Loaded libSDL3.so.0, got 0x560297f01690
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x560297c370b0
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x560297f585e0
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x560297f59610
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x560297fd3230
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x560297fcede0
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x560297fe7380
Loaded /home/******/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x560297feed80
Fossilize INFO: Overriding serialization path: "/home/******/.local/share/Steam/steamapps/shadercache/570/fozpipelinesv6/steamapprun_pipeline_cache".
crash_20231116223901_2.dmp[254867]: Uploading dump (out-of-process)
/tmp/dumps/crash_20231116223901_2.dmp
./dota.sh: line 122: 254633 Segmentation fault      (core dumped) ${STEAM_RUNTIME_PREFIX} ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
```

**Comment by alou-S** [comment]
Anyway I have finally gotten some free time and put together a very half assed workaround for the problem. 

Replacing the following libraries within the `path_to_steam_library/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64` directory from the previous Manifest (6806193462375356812) of the Dota 2 Linux Depot (373306) works around the problem.

```
MD5SUM                            Filename
946219b9037ce9ec837e3de9a1da1000  libSDL3.so.0
6c7c4f3374b89ca3cfdbcc3985af83a6  libinputsystem.so
c50af4defef575029f6e4f7feb3aa330  librendersystemvulkan.so
d7d00fff97d13637f445565ec3feb64a  libsoundsystem.so
6ee561dc20e01309496fb49304303751  libtier0.so
```

Doing this is somewhat complicated and **is not for the average user**.
Here is a brief:
1. Enable the Steam Console
2. Download the particular Manifest ID of the depot by using this command in the Steam Console : `download_depot 570 373306 6806193462375356812`
3. The downloaded files can be found in `$HOME/.steam/steam/ubuntu12_32/steamapps/content/app_570/depot_373306/`
4. Copy over the said libraries to the game directory (Please do not under any circumstance replace your system libraries with these)

Or if you trust me (**which you generally shouldn't**) you could use this archive that contains the older version of said libraries: [libs.zip](https://github.com/ValveSoftware/Dota-2/files/13377606/libs.zip)

Also hello Pedro, nice seeing you here :)

**Comment by n0ne-st4hp** [comment]
> Anyway I have finally gotten some free time and put together a very half assed workaround for the problem.
> 
> Replacing the following libraries within the `path_to_steam_library/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64` directory from the previous Manifest (6806193462375356812) of the Dota 2 Linux Depot (373306) works around the problem.


Heyo alou-S, 
thanks for provided workaround. It works on Debian Trixie, 6.5.0-4-amd64, game launched.

**Comment by chiendo1010** [comment]
Same problem on Ubuntu 20.04.6 LTS.

**Comment by TaPO4eg3D** [comment]
@alou-S it works, thanks for you time and efforts

**Comment by Niflheimr-dog** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

Thank you bro, you are our savior))))

**Comment by lwjkl** [comment]
@alou-S cheers, it works on Ubuntu 22.04.3 LTS. Thanks. I trust you.

**Comment by gentuser** [comment]
it works with  the trusted zip lib!

**Comment by RanGojo** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

Thank you, your solution works for me:

```bash
cd $HOME/.steam/steam
cp ./ubuntu12_32/steamapps/content/app_570/depot_373306/game/bin/linuxsteamrt64/libSDL3.so.0  ./steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64/libSDL3.so.0
cp ./ubuntu12_32/steamapps/content/app_570/depot_373306/game/bin/linuxsteamrt64/libinputsystem.so   ./steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64/libinputsystem.so
cp ./ubuntu12_32/steamapps/content/app_570/depot_373306/game/bin/linuxsteamrt64/librendersystemvulkan.so    ./steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64/librendersystemvulkan.so
cp ./ubuntu12_32/steamapps/content/app_570/depot_373306/game/bin/linuxsteamrt64/libsoundsystem.so     ./steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64/libsoundsystem.so
cp ./ubuntu12_32/steamapps/content/app_570/depot_373306/game/bin/linuxsteamrt64/libtier0.so      ./steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64/libtier0.so
```

**Comment by slovak12345** [comment]
It's also works for me

**Comment by root-reboot** [comment]
Ubuntu 22.04.3 LTS Wayland it's work

**Comment by manolomono** [comment]
Ok,  Thank you for libs .so Its secure this md5sum ?? jeje. Woking fine in 
![image](https://github.com/ValveSoftware/Dota-2/assets/1908613/7b3544d9-60c2-4a45-9f10-6f1ed79b93e1)

**Comment by robertpro** [comment]
> Or if you trust me (**which you generally shouldn't**) you could use this archive that contains the older version of said libraries: [libs.zip](https://github.com/ValveSoftware/Dota-2/files/13377606/libs.zip)

I trust you, and if you trust me, you can run this code in your terminal to fix the problem:

```shell
cd "$HOME/.steam/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64"
wget https://github.com/ValveSoftware/Dota-2/files/13377606/libs.zip  # This is from @alou-S 
unzip -o libs.zip 
rm -f libs.zip
```

Done!

**Comment by earn9** [comment]
@alou-S Thank you for solution, Worked well on POPOS.
GL HF

**Comment by IceWolf0110** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

thanks, this worked but you should add another script for flatpak user too

**Comment by slightlyfaulty** [comment]
This is why I love the Linux community. They'll find and fix issues before the devs manage to get out of bed. Thanks @alou-S, you're a legend!

**Comment by vincedioquino** [comment]
@alou-S Thank you, worked great. Pop-OS 22.04, Razer Blade 14 (2021).

For some reason I can't enable console (Flatpak user here), so I just copied the libs.zip file to the folder (after some manual searching).

> I was at a point of almost giving up and installing Windows in desperation lmao

**Comment by Fristivan** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

Thanks a lot! libs.zip worked on Kde neon!

**Comment by daIuvenis** [comment]
1. write into your termnal steam steam://nav/console
2. After this command will open steam with console
3. Enter this command in your steam console download_depot 570 373306 6806193462375356812. File will save by this path +- $HOME/.steam/steam/ubuntu12_32/steamapps/content/app_570/depot_373306/ 
4. Downloaded files replace with your files ~/.steam/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64

**Comment by raulnovella** [comment]
El mismo problema

**Comment by robertkirkman** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

Wow I am amazed but you were actually right from the start. Here's my way of writing what I did to follow your steps:

Bash:
```
steam steam://open/console
```
Steam:
```
download_depot 570 373306 6806193462375356812
```
Bash:
```
cd $HOME/.steam/steam/ubuntu12_32/steamapps/content/app_570/depot_373306/game/bin/linuxsteamrt64/
cp libSDL3.so.0 /mnt/ATOM/SteamLibrary/steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64
cp libinputsystem.so /mnt/ATOM/SteamLibrary/steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64
cp librendersystemvulkan.so /mnt/ATOM/SteamLibrary/steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64
cp libsoundsystem.so /mnt/ATOM/SteamLibrary/steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64
cp libtier0.so /mnt/ATOM/SteamLibrary/steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64
```
**How did you know, specifically, that these were the files necessary to replace**?

**Comment by josephlett** [comment]
> @josephlett When quoting a reply you don't need to quote the entire reply. Just the first 2 to three lines. It would be nice if you could edit your reply and remove the extra lines. You could probably instead add a link to the reply along with the first few lines for reference

Sry about it thanks for the advice
Was not trying to cause spam. 
Sry please I apologise

**Comment by pavildon** [comment]
> [#2603 (comment)](https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378)

I'm wandering if this fix can get you in trouble, because integrity checks that the game might do and get you banned for messing with the files.

**Comment by alou-S** [comment]
> > Replying to [#2603 (comment)](https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378)
> **How did you know, specifically, that these were the files necessary to replace**?

Looked at the change lists and took a few guesses based on error message. I had already known its something related to SDL early on. But simply replacing the SDL library didn't work so I took a few guesses.

**Comment by alou-S** [comment]
> > [#2603 (comment)](https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378)
> 
> I'm wandering if this fix can get you in trouble, because integrity checks that the game might do and get you banned for messing with the files.

Very extremely unlikely due to how VAC works based on recognizing known cheat signatures. So its pretty much impossible for VAC to flag valve's own libraries

**Comment by gentuser** [comment]
on top of that you should be reported, and if you don't use cheat, no one will report that to you

**Comment by spmsupun** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

this worked, thank you so much

**Comment by pixel1001** [comment]
It also works for me, debian 11, nvidia2070
Thanks for the workaround

I heard that this will boost your mmr, haha

**Comment by ghost** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

I can get to splash screen with this, but the client closes after 15-20 seconds. Ubuntu 22.04 x64

**Comment by timmac41** [comment]
> Anyway I have finally gotten some free time and put together a very half assed workaround for the problem.
> 
> Replacing the following libraries within the `path_to_steam_library/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64` directory from the previous Manifest (6806193462375356812) of the Dota 2 Linux Depot (373306) works around the problem.
> 

Fixes my issues on KDE/Wayland/AMD/Arch. Just played a turbo. Thank you so much

**Comment by drmmwlt** [comment]
thanks @alou-S @robertpro and others ! ! !  that fixed it on manjaro nvidia !

**Comment by WhitleyStriber** [comment]
Have the same issue on Gentoo.

[log.txt](https://github.com/ValveSoftware/Dota-2/files/13380155/log.txt)

Fixed with https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

Here is a script: 
[patch.txt](https://github.com/ValveSoftware/Dota-2/files/13380265/patch.txt)


TYSM!

**Comment by pchmykh** [comment]
Fedora 39, the same.

**Comment by hakantakiri** [comment]
Same error, PopOs 20.04

**Comment by hakantakiri** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

This is the preferred fix, just worked in Pop_OS 20.04. Please use this since it references official Dota 2 files. Avoid directly downloading another zip file from another answer.

**Comment by pixel1001** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814940834

I believe the best reference its that you download the libraries from steam console...
download_depot 570 373306 6806193462375356812

and not from other source, just if you trust on [alou-S](https://github.com/alou-S)...
:P

**Comment by nedosekinstanislav** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

Thanks, I went to feed my opponents for this update

**Comment by Turmfalke2** [comment]
Can confirm the workaround of using the old so works. Probably a good idea to script the copying part so you can rerun it the next time the game updates.

Also is there a way to permanently enable the console? Like a config setting?

**Comment by denisoct** [comment]
> Can confirm the workaround of using the old so works. Probably a good idea to script the copying part so you can rerun it the next time the game updates.
> 
> Also is there a way to permanently enable the console? Like a config setting?

`-console` in steam.desktop file

`Exec=/usr/bin/steam %U -console`

**Comment by alou-S** [comment]
> Can confirm the workaround of using the old so works. Probably a good idea to script the copying part so you can rerun it the next time the game updates.
> 
> Also is there a way to permanently enable the console? Like a config setting?

No point in scripting this, the next time the game updates those library files will be fixing the libraries.

**Comment by chiendo1010** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

Thank you! It worked on Ubuntu 20.04.6 LTS

**Comment by gperevalov** [comment]
@alou-S Ubuntu 22.04.3 LTS, nvidia, it worked! Thx <3

**Comment by SanyaBurmantov** [comment]
@robertpro

thank you so much! It works

**Comment by AriePriyambadha** [comment]
@alou-S Work perfectly on my ubuntu thank you

**Comment by Hurr1** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378

thx bro Nvidia Ubuntu 22.04.3 LTS

**Comment by danieljennings** [comment]
Thank you for your reports, we believe we understand what's happening here and are preparing a fix.

**Comment by petry078** [comment]
> ```shell
> cd "$HOME/.steam/steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64"
> wget https://github.com/ValveSoftware/Dota-2/files/13377606/libs.zip  # This is from @alou-S 
> unzip -o libs.zip 
> rm -f libs.zip
> ```

Great. it works!

**Comment by RivalAtDW** [comment]
TY, @alou-S (manjaro linux nvidia 3050ti)

**Comment by seixasxbr** [comment]
same issue

**Comment by sylware** [comment]
segfaulting here too (and zenity is not even working due to some missing gtk3 symbol in the official steam runtime).

Will wait for the fix as I have to already replace libSDL3 in order to get alsa sound (it is under heavy dev...)

**Comment by Turmfalke2** [comment]
> > Can confirm the workaround of using the old so works. Probably a good idea to script the copying part so you can rerun it the next time the game updates.
> > Also is there a way to permanently enable the console? Like a config setting?
> 
> No point in scripting this, the next time the game updates those library files will be fixing the libraries.

how are you so certain of that?

**Comment by UtkarshDixit-97** [comment]
> > how are you so certain of that?

A steam dev @danieljennings just stated that they aare working on the fix, he commented 1 hour ago. Hopefully it gets fixed, I haven't run the fix mentioned earlier for the same reason, I am hoping for steam to provide real patch.

**Comment by pretender07** [comment]
me sale el mismo error en Fedora 39, por un momento pensé que era la nueva versión de la distro

**Comment by snorlax212** [comment]
Same issue. Xubuntu 22.04.

**Comment by hashwagon** [comment]
I have this problem and BTW I use Arch.

```
                   -`                    sniper@midorfeed
                  .o+`                   ---------------- 
                 `ooo/                   OS: Arch Linux x86_64 
                `+oooo:                  Host: ROG Zephyrus G14 GA402RJ_GA402RJ 1.0 
               `+oooooo:                 Kernel: 6.6.1-arch1-1 
               -+oooooo+:                Uptime: 2 years, 305 days, 6 hours, 14 mins 
             `/:-:++oooo+:               Packages: 1326 (pacman) 
            `/++++/+++++++:              Shell: bash 5.2.15 
           `/++++++++++++++:             Resolution: 3840x2160 
          `/+++ooooooooooooo/`           DE: Plasma 5.27.9 
         ./ooosssso++osssssso+`          WM: kwin 
        .oossssso-````/ossssss+`         Theme: [Plasma], Breeze [GTK2/3] 
       -osssssso.      :ssssssso.        Icons: [Plasma], breeze-dark [GTK2/3] 
      :osssssss/        osssso+++.       Terminal: konsole 
     /ossssssss/        +ssssooo/-       Terminal Font: Source Code Pro 12 
   `/ossssso+/:-        -:/+osssso+-     CPU: AMD Ryzen 9 6900HS with Radeon Graphics (16) @ 4.935GHz 
  `+sso+:-`                 `.-/+oso:    GPU: AMD ATI Radeon RX 6650 XT / 6700S / 6800S 
 `++:.                           `-/+/   GPU: AMD ATI Radeon 680M 
 .`                                 `/   Memory: 8724MiB / 39334MiB
 ```

**Comment by Emanem** [comment]
Have same issue, applied [workaround](https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814368378) to replace _.so_ and played a ranked match without problems.

**Comment by denisoct** [comment]
> me sale el mismo error en Fedora 39, por un momento pensé que era la nueva versión de la distro

¿Probaste esto? https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

**Comment by abrahamlnx** [comment]
> Have the same issue on Gentoo.
> 
> [log.txt](https://github.com/ValveSoftware/Dota-2/files/13380155/log.txt)
> 
> Fixed with [#2603 (comment)](https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522)
> 
> Here is a script: [patch.txt](https://github.com/ValveSoftware/Dota-2/files/13380265/patch.txt)
> 
> TYSM!

ty for script

**Comment by zargex** [comment]
https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522 
This solution worked for me :+1:

**Comment by Everest10** [comment]
Same issue here, fedora 39 kde plasma 5.27.9 Wayland 
![image](https://github.com/ValveSoftware/Dota-2/assets/7228207/8a59a984-7a47-411a-a106-429d44e0bb87)

**Comment by mysticlizard** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2603#issuecomment-1814538522

This chaotic workaround is successful for me.

**Comment by ErodeesFleurs** [comment]
Same issue. Manjaro.

**Comment by Chaliuz** [comment]
> Anyway I have finally gotten some free time and put together a very half assed workaround for the problem.
> 
> Replacing the following libraries within the `path_to_steam_library/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64` directory from the previous Manifest (6806193462375356812) of the Dota 2 Linux Depot (373306) works around the problem.
> 
> ```
> MD5SUM                            Filename
> 946219b9037ce9ec837e3de9a1da1000  libSDL3.so.0
> 6c7c4f3374b89ca3cfdbcc3985af83a6  libinputsystem.so
> c50af4defef575029f6e4f7feb3aa330  librendersystemvulkan.so
> d7d00fff97d13637f445565ec3feb64a  libsoundsystem.so
> 6ee561dc20e01309496fb49304303751  libtier0.so
> ```
> 
> Doing this is somewhat complicated and **is not for the average user**. Here is a brief:
> 
> 1. Enable the Steam Console
> 2. Download the particular Manifest ID of the depot by using this command in the Steam Console : `download_depot 570 373306 6806193462375356812`
> 3. The downloaded files can be found in `$HOME/.steam/steam/ubuntu12_32/steamapps/content/app_570/depot_373306/`
> 4. Copy over the said libraries to the game directory (Please do not under any circumstance replace your system libraries with these)
> 
> Or if you trust me (**which you generally shouldn't**) you could use this archive that contains the older version of said libraries: [libs.zip](https://github.com/ValveSoftware/Dota-2/files/13377606/libs.zip)
> 
> Also hello Pedro, nice seeing you here :)

Thanks bro, this worked for me :3

**Comment by See-Night** [comment]
I've found solutions from [here](https://steamcommunity.com/app/570/discussions/0/6027566653451436967/?ctp=10), but there's no guarantee they'll work for everyone:

```bash
cd $HOME/.steam/steam/steamapps/common/dota\ 2\ beta/game/bin/linuxsteamrt64
wget https://github.com/ValveSoftware/Dota-2/files/13377606/libs.zip
unzip libs.zip && rm libs.zip
```

**Comment by faroperfect** [comment]
time 2 close this issue, the latest patch have fixed this problem.

**Comment by nhtua** [comment]
lol, unbelievable the same issue appears again after 6 years 
https://github.com/ValveSoftware/Dota-2-Vulkan/issues/153

**Comment by alexeysvrv** [comment]
@danieljennings @slouken Tell me, is there any progress in porting the Source 2 engine and the Dota 2 game to Wayland? At least some information, please

**Comment by cdonatom** [comment]
@slouken this issue is still present for AMD GPUs and Mesa 23.2

**Comment by slouken** [comment]
Please open a new issue for AMD GPUs and Mesa 23.2. It's likely that the combination listed actually doesn't have the Vulkan support DOTA needs.

**Comment by formidabilus** [comment]
it's back again and the script from above doesn't work anymore. Fedora 39, nvidia 3070TI
